.github/workflows/basic.yml

```diff
@@ -96,48 +110,80 @@ jobs:
       working-directory:  ${{github.workspace}}
       run: |
         source $GITHUB_WORKSPACE/.github/scripts/install_hadoop.sh
-        SPARK_ENV=$SPARK_ENV source $GITHUB_WORKSPACE/.github/scripts/install_spark.sh
+        JAVA_HOME=$JAVA_LINUX SPARK_ENV=$SPARK_ENV source $GITHUB_WORKSPACE/.github/scripts/install_spark.sh
       env:
         INSTALL_DIR: ${{runner.workspace}}
 
     - name: Configure CMake Build ubuntu
       shell: bash
-      if: startsWith(matrix.os, 'ubuntu')
+      if: startsWith(matrix.os, 'ubuntu') && matrix.java == '8'
+      working-directory: ${{env.GENOMICSDB_BUILD_DIR}}
+      run: |
+        source $PREREQS_ENV
+        JAVA_HOME=$JAVA_LINUX cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=Coverage -DCMAKE_INSTALL_PREFIX=$CMAKE_INSTALL_PREFIX -DCMAKE_PREFIX_PATH=$PREREQS_INSTALL_DIR -DGENOMICSDB_RELEASE_VERSION=$GENOMICSDB_RELEASE_VERSION -DBUILD_JAVA=1
+
+    - name: Configure CMake Build ubuntu Java 11
```

Comment:
Can you combine "Configure CMake Build ubuntu Java 11" with "Configure CMake Build ubuntu". Looks like they only differ in a few ways, so you could add a bash if conditional:
```
if [[ ${{ matrix.java }} == '11' ]]; then
    export JAVA_HOME=$JAVA_LINUX
    echo $JAVA_LINUX/bin >> $GITHUB_PATH
    export GDB_MAVEN_PROFILE=java11
fi
source $PREREQS_ENV
cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=Coverage -DCMAKE_INSTALL_PREFIX=$CMAKE_INSTALL_PREFIX -DCMAKE_PREFIX_PATH=$PREREQS_INSTALL_DIR -DGENOMICSDB_RELEASE_VERSION=$GENOMICSDB_RELEASE_VERSION -DBUILD_JAVA=1 -DGENOMICSDB_MAVEN_PROFILE=$GDB_MAVEN_PROFILE
```
Not sure if the exports will work in there or if you need to pull them out into a conditional `env` or something...

Reply:
Thanks for the suggestion, I've combined a few steps using this strategy. The build for that (containing mac java 11) is confirmed here: https://github.com/GenomicsDB/GenomicsDB/actions/runs/496940810

Reply:
Weird, the PR compile crashed, but the idential push to this branch is still running fine: https://github.com/GenomicsDB/GenomicsDB/runs/1730943441?check_suite_focus=true